core/typedefs.h

```diff
@@ -109,7 +109,8 @@ constexpr T ABS(T m_v) {
 
 template <typename T>
 constexpr const T SIGN(const T m_v) {
-	return m_v == 0 ? 0.0f : (m_v < 0 ? -1.0f : +1.0f);
+	// Sign of NAN is NAN, sign of -0.0 is -0.0.
+	return m_v > 0 ? +1.0f : (m_v < 0 ? -1.0f : m_v);
```

Comment:
```suggestion
	return m_v > 0 ? +1.0f : (m_v < 0 ? -1.0f : 0.0f);
```
With this change, and the appropriate changes to the docs, I'd approve this, as I agree that returning `+1` doesn't make sense for `NaN`